src/Orchard.Web/Modules/Orchard.DynamicForms/Views/EditorTemplates/FormBindings.cshtml

```diff
@@ -36,7 +36,7 @@
                 }
                 @foreach (var fieldDescriptor in partDescriptor.FieldBindings) {
                     var fieldBindingIndex = 0;
-                    var fieldState = partState.Fields.ElementAtOrDefault(fieldIndex) ?? new FieldBindingSettings();
+                    var fieldState = partState.Fields.FirstOrDefault(f=>f.Name == fieldDescriptor.Field.Name) ?? new FieldBindingSettings();                    
```

Comment:
Only the first time the bindings screen loads, but then a new instance is created automatically. But I'll review this code in detail tomorrow.